### PR TITLE
Add support for -scala-output-version flag

### PR DESCRIPTION
**Pull request by prolativ**
WIP

### CODE REVIEW

**Comment by lefou** [comment]
It would be nice, if you could update the PR description with a short summary of what this PR does and probably a link to the respective feature in the Scala compiler. Can you give a rough estimate, when this feature is going to be consumable in scalac and hence this PR should get it? Maybe some ticket we can link to?

**Comment by lefou** [comment]
* https://github.com/lampepfl/dotty/pull/14606

File: scalalib/src/JavaModule.scala [hunk 1]
```diff
@@ -357,6 +357,11 @@ trait JavaModule
     })()
   }
 
+  /**
+   * Like [[resolvedIvyDeps]] but used only for compiling this module .
+   */
+  def resolvedCompilationIvyDeps: T[Agg[PathRef]] = resolvedIvyDeps
```
Comment (lefou) [inline scalalib/src/JavaModule.scala:363]: I'm not that happy with this name. First, there is a high change of reading it as `resolveCompileIvyDeps` which would be the dual of `compileIvyDeps` (which currently does not exists). Also, the scaladoc gives no clear idea, why this is needed exactly and what's the difference to `resolvedIvyDeps`.

File: scalalib/src/ScalaModule.scala [hunk 2]
```diff
@@ -57,6 +58,8 @@ trait ScalaModule extends JavaModule { outer =>
    */
   def scalaVersion: T[String]
 
+  def scalaOutputVersion: T[String] = scalaVersion
```
Comment (lefou) [inline scalalib/src/ScalaModule.scala:61]: Here we need some scaladoc explaining its purpose and also the distinction to `scalaVersion`.

**Comment by lolgab** [comment]
Scala 3.1.2 was released and the companion Sbt PR was merged.
https://github.com/sbt/sbt/pull/6814

**Comment by lolgab** [comment]
Since the compiler team decided to drop the `-scala-output-version` flag we don't need to support it anymore.

**Pull request closed without merging**